Ext/Accessibility/src/main/java/dev/testify/accessibility/internal/BaselineGenerator.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2022 ndtp
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package dev.testify.accessibility.internal
+
+import android.content.Context
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import dev.testify.TestDescription
+import dev.testify.accessibility.exception.MalformedBaselineException
+import dev.testify.internal.DeviceIdentifier
+import dev.testify.internal.helpers.loadAsset
+import dev.testify.internal.output.OutputFileUtility
+import dev.testify.testDescription
+import java.io.File
+import java.io.InputStreamReader
+
+/**
+ * Write the a11y.json file for the current test.
+ */
+internal fun writeAccessibilityCheckResults(results: CheckResults) {
+    val description = getInstrumentation().testDescription
+    val context = getInstrumentation().targetContext
+    getOutputFile(context, description).writeText(results.toJson())
+}
+
+/**
+ * Read the a11y.json file for the current test.
+ */
+internal fun readAccessibilityCheckResults(context: Context): CheckResults? {
+    val description = getInstrumentation().testDescription
+    val filePath = OutputFileUtility().getFileRelativeToRoot(
+        subpath = DeviceIdentifier.getDescription(context),
+        fileName = description.name,
+        extension = JSON_EXTENSION
+    )
```

Comment:
I'm a bit confused why the file resolution is different for the read and write cases? Any reason we couldn't do:

```kt
val rawJson = getOutputFile(context, description).readText()
```

Reply:
Oh, yes. The reason this is different is because we `read` from the `assets` directory packaged in & deployed with the APK.
For the `write`, we write the file to `data/data` (or the sdcard, depending on configuration).

We can't/don't want to write anything to `assets` as those are the pristine baselines.

@gantonious